utilities/backupable/backupable_db.cc

```diff
@@ -354,14 +411,15 @@ class BackupEngineImpl : public BackupEngine {
       const std::string& src, const std::string& dst,
       const std::string& contents, Env* src_env, Env* dst_env,
       const EnvOptions& src_env_options, bool sync, RateLimiter* rate_limiter,
-      uint64_t* size = nullptr, uint32_t* checksum_value = nullptr,
-      uint64_t size_limit = 0,
+      const std::string& src_checksum_func_name, uint64_t* size = nullptr,
+      uint32_t* checksum_value = nullptr, uint64_t size_limit = 0,
       std::function<void()> progress_callback = []() {},
       std::string* db_id = nullptr, std::string* db_session_id = nullptr);
 
-  Status CalculateChecksum(const std::string& src, Env* src_env,
-                           const EnvOptions& src_env_options,
-                           uint64_t size_limit, uint32_t* checksum_value);
+  Status CalculateChecksum(
+      const std::string& src, Env* src_env, const EnvOptions& src_env_options,
+      uint64_t size_limit, uint32_t* checksum_value,
```

Comment:
I have one question here. If the FileChecksumGenerator provided by user does not provide the uint32_t type checksum results (e.g., SHA1 or MD5), is there any issues here?

Reply:
Good question, I think that would be problematic. It seems better to use `string` for storing any checksum result (default or custom). In that case, one more thing to worry about is that the custom checksum result may contain non-printable content or characters not allowed to appear in the filename.

Reply:
If we decide to always use the backup engine default checksum and treat the custom checksum as an extra protection, we may simply use the default checksum in the filename so that we do not need to worry about non-printable content or disallowed characters. (perhaps still need to take care of the case that some raw bytes may be converted to newline characters when storing checksum results in backup meta files?)